### PR TITLE
Implemented the MVVM architecture

### PR DESCRIPTION
**Pull request by IanisT227**

### CODE REVIEW

File: app/src/main/java/com/internship/move/feature/map/MapFragment.kt [hunk 1]
```diff
@@ -25,10 +33,9 @@ class MapFragment : Fragment(R.layout.fragment_map) {
         }
 
         binding.clearBtn.setOnClickListener {
-            requireActivity().getPreferences(Context.MODE_PRIVATE)
-                .edit()
-                .putBoolean("IS_LOGGED", false)
-                .apply()
+            scope.launch {
```
Comment (tapp-hadaralex) [inline app/src/main/java/com/internship/move/feature/map/MapFragment.kt]: better use `viewModelScope.launch`. The view might get destroyed faster and the scope cancelled.
Move it in the viewmodel

File: app/src/main/java/com/internship/move/feature/onboarding/OnboardingFragment.kt [hunk 2]
```diff
@@ -34,7 +42,7 @@ class OnboardingFragment : Fragment(R.layout.fragment_onboarding) {
                 }
                 binding.onboardingVP2.currentItem = binding.onboardingVP2.currentItem + 1
             } else {
-                initPersistence()
+                scope.launch { viewModel.logIn() }
```
Comment (tapp-hadaralex) [inline app/src/main/java/com/internship/move/feature/onboarding/OnboardingFragment.kt]: Try something new: Have a liveData that holds a boolean value that indicates if the user's logged in and observe it.

File: app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt [hunk 3]
```diff
@@ -21,11 +30,13 @@ class SplashFragment : Fragment(R.layout.fragment_splash) {
     }
 
     private fun navigateToNextFragment() {
-        val pref = requireActivity().getPreferences(Context.MODE_PRIVATE)
-        if (pref.getBoolean("IS_LOGGED", false)) {
-            findNavController().navigate(SplashFragmentDirections.actionGlobalRegisterFragment())
-        } else {
-            findNavController().navigate(SplashFragmentDirections.actionSplashFragmentToOnboardingFragment())
+
+        scope.launch {
```
Comment (tapp-hadaralex) [inline app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt]: Do like in the Onboarding fragment here too!

**Review by tapp-hadaralex — CHANGES_REQUESTED** [review]
Really nice PR! Congrats!

Try to implemented my suggestions; might make the flow easier

File: app/src/main/java/com/internship/move/feature/authentication/register/RegisterFragment.kt [hunk 4]
```diff
@@ -64,6 +64,7 @@ class RegisterFragment : Fragment(R.layout.fragment_register) {
         } else {
             binding.launchHomeBtn.isEnabled = false
             binding.launchHomeBtn.setTextColor(ResourcesCompat.getColor(resources, R.color.neutral_pink, null))
+
```
Comment (catadarjan-tapptitude) [inline app/src/main/java/com/internship/move/feature/authentication/register/RegisterFragment.kt:67]: Extra empty line :P

File: app/src/main/java/com/internship/move/InternalStorageManager.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package com.internship.move
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+
+class InternalStorageManager(context: Context) {
+    private val preferences: SharedPreferences = context.getSharedPreferences(KEY_IS_LOGGED, MODE_PRIVATE)
```
Comment (catadarjan-tapptitude) [inline app/src/main/java/com/internship/move/InternalStorageManager.kt]: Leave empty line between calss declaration and first field/method

**Review by catadarjan-tapptitude — COMMENTED** [review]
Just a couple small comments
Nice job 👏

**Pull request merged**